diff --git third_party/openvr/src/src/vrcommon/vrpathregistry_public.cpp third_party/openvr/src/src/vrcommon/vrpathregistry_public.cpp
index 137b36f57cd4..dcbb9aaed52f 100644
--- third_party/openvr/src/src/vrcommon/vrpathregistry_public.cpp
+++ third_party/openvr/src/src/vrcommon/vrpathregistry_public.cpp
@@ -1,7 +1,11 @@
 //========= Copyright Valve Corporation ============//
 
 #include "vrpathregistry_public.h"
-#include "json/json.h"
+
+#include <json/reader.h>
+#include <json/value.h>
+#include <json/writer.h>
+
 #include "pathtools_public.h"
 #include "envvartools_public.h"
 #include "strtools_public.h"
