diff --git content/browser/renderer_host/legacy_render_widget_host_win.cc content/browser/renderer_host/legacy_render_widget_host_win.cc
index 30b95a5dcd0b..bf9d308879cc 100644
--- content/browser/renderer_host/legacy_render_widget_host_win.cc
+++ content/browser/renderer_host/legacy_render_widget_host_win.cc
@@ -342,7 +342,7 @@ LRESULT LegacyRenderWidgetHostHWND::OnMouseLeave(UINT message,
     // has moved outside the bounds of the parent.
     POINT cursor_pos;
     ::GetCursorPos(&cursor_pos);
-    if (::WindowFromPoint(cursor_pos) != GetParent()) {
+    if (::WindowFromPoint(cursor_pos) != hwnd()) {
       bool msg_handled = false;
       ret = GetWindowEventTarget(GetParent())->HandleMouseMessage(
           message, w_param, l_param, &msg_handled);
